app/src/main/java/co/condorlabs/customcomponents/graphics/barsgraph/BarsGraph.kt

```diff
@@ -0,0 +1,266 @@
+        val viewMeasuredHeight = this.measuredHeight
+
+        // DRAW HORIZONTAL LINE
+        val horizontalLinePositionY = getPixelsValueWithPercentage(viewMeasuredHeight, 50F)
+        val horizontalLinesMarginWithParenTop =
+            getPixelsValueWithPercentage(viewMeasuredHeight, 10F)
+        val horizontalLinesSpacingInPixels =
+            (horizontalLinePositionY - horizontalLinesMarginWithParenTop) / numberOfHorizontalLines
+        val horizontalLineStartPositionX = getPixelsValueWithPercentage(viewMeasuredWidth, 10F)
+        val horizontalLineEndPositionX = getPixelsValueWithPercentage(viewMeasuredWidth, 90F)
+
+        canvas.drawLine(
+            horizontalLineStartPositionX,
+            horizontalLinePositionY,
+            horizontalLineEndPositionX,
+            horizontalLinePositionY,
+            horizontalLinesPaint
+        )
+
+        // DRAW BARS
+        barsGraphConfig?.let { chartConfig ->
+            barsStrokeWidthMiddle = barsStrokeWidth / 2
+            val pixelRangeToDrawBars = horizontalLinesSpacingInPixels * (numberOfHorizontalLines)
+            val strokeWidthPercentageValueOfBars =
+                ((barsStrokeWidth + barsMargin) * 100) / viewMeasuredWidth
+            val barsSpacingInPixels = getPixelsValueWithPercentage(
+                viewMeasuredWidth,
+                widthPercentageToDrawLinesAndBars - strokeWidthPercentageValueOfBars
+            ) / (numberOfBars - 1)
+            var barPositionX = getPixelsValueWithPercentage(viewMeasuredWidth, 10F)
+            val barMarginOffset = (barsStrokeWidth + barsMargin) / 2
+            val labelsPositionY = getPixelsValueWithPercentage(viewMeasuredHeight, 60F)
+            val countLabelsPositionY = getPixelsValueWithPercentage(viewMeasuredHeight, 85F)
+            val barStartPositionX = getPixelsValueWithPercentage(viewMeasuredHeight, 50F)
+
+            chartConfig.bars.forEach { bar ->
+                val totalSumOfBarSections = bar.value
+                val barPositionXWithOffset = barPositionX + barMarginOffset
+                val barPercentageToDraw = getBarHeightPercentage(totalSumOfBarSections.toFloat())
+                val pixelsToSubtractFromY =
+                    getPixelsValueWithPercentage(pixelRangeToDrawBars.toInt(), barPercentageToDraw)
+                val barStartPositionY = barStartPositionX - pixelsToSubtractFromY
+
+                canvas.drawPath(
+                    Path().apply {
+                        addRoundRect(
+                            RectF(
+                                barPositionXWithOffset - barsStrokeWidthMiddle,
+                                barStartPositionY,
+                                barPositionXWithOffset + barsStrokeWidthMiddle,
+                                barStartPositionX
+                            ), corners, Path.Direction.CW
+                        )
+                    },
+                    barsPaint.apply { color = bar.strokeColor }
```

Comment:
break down into small methods, this could be 1 method

Reply:
Ok